src/Task.h

```diff
@@ -864,10 +870,32 @@ class Task {
     return was_reaped;
   }
 
+  void did_reap() {
+    was_reaped = true;
+  }
+
   void os_exec_stub(SupportedArch arch);
 
+  /**
+   * Try to make the current task look exactly like some `other` task
+   * by copying that task's address space and other relevant properties,
+   * but without using the os's clone system call.
+   */
+  void dup_from(Task *task);
```

Comment:
Normally a `Task` is an rr tracee or a diversion spawned from it. Here it's something we don't want rr to manage. Are you using `Task` and `DiversionSession` here just because they provide handy utilities for working with the subprocess?

Reply:
Yes, though rr does still manage the Task until we fully detach from it, so I think having it be a `Task` is fine. DiversionSession is indeed a bit of a pun though.